File: src/main/java/pl/training/workshop/m6/s07_decorator/start/TicketAssembler.java
package pl.training.workshop.m6.s07_decorator.start;

import pl.training.workshop.m6.s07_decorator.TicketOrder;

/** Start: jedyne miejsce tworzenia biletu - tu później złożymy łańcuch dekoratorów. */
public final class TicketAssembler {
    public Ticket assemble(TicketOrder order) {
        return new Ticket(order);
    }
}

File: csharp/src/Training.Workshop/M6/S07Decorator/Step2/TicketAssembler.cs
namespace Training.Workshop.M6.S07Decorator.Step2;

/// <summary>Krok 2: fabryka składa łańcuch - rdzeń, a na zewnątrz opcjonalne ubezpieczenie.</summary>
public sealed class TicketAssembler
{
    public IPricedTicket Assemble(TicketOrder order)
    {
        IPricedTicket ticket = new Ticket(order);
        if (order.Insurance)
        {
            ticket = new Insurance(ticket);
        }
        return ticket;
    }
}
